manifests/config.pp

```diff
@@ -31,9 +42,10 @@
           mode    => '0644',
           owner   => 'root',
           group   => 'root',
+          notify  => $systemd_notify,
           content => template('prometheus/prometheus.systemd.erb'),
         }
-        ~> exec { 'prometheus-systemd-reload':
+        exec { 'prometheus-systemd-reload':
```

Comment:
could you replace this exec with a notify to the one provided here: https://github.com/camptocamp/puppet-systemd

and add the module as dependency to the metadata.json